### ISSUE TITLE
SQL Syntax error when accessing to blog comment in backend

### ISSUE THREAD

**Issue by GhaziTriki** [forkcms/forkcms#1663]
## Type

- Critical bug

## Problem description

```
SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #6 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'forkcms.p.title' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

500 Internal Server Error - PDOException

Stack Trace

in vendor/spoon/library/spoon/database/database.php at line 484   -
        }
        // execute statement
        $statement->execute();
        // has errors
        if($statement->errorCode() != 0)
at PDOStatement ->execute () 
in vendor/spoon/library/spoon/database/database.php at line 484   +
at SpoonDatabase ->getNumRows ('SELECT i.id, UNIX_TIMESTAMP(i.created_on) AS created_on, i.author, i.text, p.id AS post_id, p.title AS post_title, m.url AS post_url FROM blog_comments AS i INNER JOIN blog_posts AS p ON i.post_id = p.id AND i.language = p.language INNER JOIN meta AS m ON p.meta_id = m.id WHERE i.status = ? AND i.language = ? AND p.status = ? GROUP BY i.id', array('published', 'en', 'active')) 
in vendor/spoon/library/spoon/datagrid/source_db.php at line 152   +
at SpoonDatagridSourceDB ->setNumResults () 
in vendor/spoon/library/spoon/datagrid/source_db.php at line 186   +
at SpoonDatagridSourceDB ->setQuery (array('SELECT i.id, UNIX_TIMESTAMP(i.created_on) AS created_on, i.author, i.text, p.id AS post_id, p.title AS post_title, m.url AS post_url FROM blog_comments AS i INNER JOIN blog_posts AS p ON i.post_id = p.id AND i.language = p.language INNER JOIN meta AS m ON p.meta_id = m.id WHERE i.status = ? AND i.language = ? AND p.status = ? GROUP BY i.id', array('published', 'en', 'active')), null) 
in vendor/spoon/library/spoon/datagrid/source_db.php at line 83   +
at SpoonDatagridSourceDB ->__construct (object(SpoonDatabase), array('SELECT i.id, UNIX_TIMESTAMP(i.created_on) AS created_on, i.author, i.text, p.id AS post_id, p.title AS post_title, m.url AS post_url FROM blog_comments AS i INNER JOIN blog_posts AS p ON i.post_id = p.id AND i.language = p.language INNER JOIN meta AS m ON p.meta_id = m.id WHERE i.status = ? AND i.language = ? AND p.status = ? GROUP BY i.id', array('published', 'en', 'active')), null) 
in src/Backend/Core/Engine/DataGridDB.php at line 33   +
at DataGridDB ->__construct ('SELECT i.id, UNIX_TIMESTAMP(i.created_on) AS created_on, i.author, i.text, p.id AS post_id, p.title AS post_title, m.url AS post_url FROM blog_comments AS i INNER JOIN blog_posts AS p ON i.post_id = p.id AND i.language = p.language INNER JOIN meta AS m ON p.meta_id = m.id WHERE i.status = ? AND i.language = ? AND p.status = ? GROUP BY i.id', array('published', 'en', 'active')) 
in src/Backend/Modules/Blog/Actions/Comments.php at line 74   +
at Comments ->loadDataGrids () 
in src/Backend/Modules/Blog/Actions/Comments.php at line 59   +
at Comments ->execute () 
in src/Backend/Core/Engine/Action.php at line 98   +
at Action ->execute () 
in src/Backend/Core/Engine/Backend.php at line 30   +
at Backend ->display () 
in app/routing.php at line 174   +
at ApplicationRouting ->handleApplication (object(Backend)) 
in app/routing.php at line 69   +
at ApplicationRouting ->backendController (object(Request), 'blog', 'comments')
at call_user_func_array (array(object(ApplicationRouting), 'backendController'), array(object(Request), 'blog', 'comments')) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 144   +
at HttpKernel ->handleRaw (object(Request), '1') 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 64   +
at HttpKernel ->handle (object(Request), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/ContainerAwareHttpKernel.php at line 69   +
at ContainerAwareHttpKernel ->handle (object(Request), '1', true) 
in app/Kernel.php at line 46   +
at Kernel ->handle (object(Request)) 
in index.php at line 63   +
Logs   -

1 error
INFO - Matched route "backend".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\DebugHandlersListener::configure".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\ProfilerListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\DumpListener::configure".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\ValidateRequestListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Bundle\FrameworkBundle\EventListener\SessionListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\RouterListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\LocaleListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\TranslatorListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Common\Mailer\Configurator::onKernelRequest".
DEBUG - Notified event "kernel.controller" to listener "Symfony\Bundle\FrameworkBundle\DataCollector\RouterDataCollector::onKernelController".
DEBUG - Notified event "kernel.controller" to listener "Symfony\Component\HttpKernel\DataCollector\RequestDataCollector::onKernelController".
INFO - Executing backend action 'Comments' for module 'Blog'.
CRITICAL - Uncaught PHP Exception PDOException: "SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #6 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'forkcms.p.title' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by" at /var/www/html/vendor/spoon/library/spoon/database/database.php line 484
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\DebugHandlersListener::configure".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\ProfilerListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\DumpListener::configure".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\ValidateRequestListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Bundle\FrameworkBundle\EventListener\SessionListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\RouterListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\LocaleListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Symfony\Component\HttpKernel\EventListener\TranslatorListener::onKernelRequest".
DEBUG - Notified event "kernel.request" to listener "Common\Mailer\Configurator::onKernelRequest".
DEBUG - Notified event "kernel.controller" to listener "Symfony\Bundle\FrameworkBundle\DataCollector\RouterDataCollector::onKernelController".
DEBUG - Notified event "kernel.controller" to listener "Symfony\Component\HttpKernel\DataCollector\RequestDataCollector::onKernelController".
```

## Steps to reproduce

Go to Modules -> Blog -> Comments in backend http://192.168.33.150/private/en/blog/comments

## Expected behavior / Proposed solutions

Should see list of comments.

**Comment by carakas** [comment]
@GhaziTriki did you make any changes ?
What fork version is it ?

The page does work on the demo
http://demo.forkcms.com/private/en/blog/comments

**Comment by carakas** [comment]
@GhaziTriki is it possible that you have `only_full_group_by` enabled on your mysql server ? Do you have some additional information for us ?

**Comment by GhaziTriki** [comment]
I have installed ForkCMS on a new Vagrant VM with PHP 7.0 and MySQL Server 5.7.

**Comment by carakas** [comment]
@GhaziTriki and the fork version is 4.1 ?

**Comment by carakas** [comment]
@GhaziTriki it is because of the option `only_full_group_by` if you disable that option you should be fine for now

**Comment by carakas** [comment]
we will include a fix for this in 4.1.1

**Comment by GhaziTriki** [comment]
I confirm the issue is with version 4.1.0 of ForkCMS

**Comment by GhaziTriki** [comment]
I confirm when disabling `only_full_group_by` and cleaning the cache blog comments backend page works perfectly.

**Comment by carakas** [comment]
fixed in #1667
